src/components/bottomSheet/BottomSheet.tsx

```diff
@@ -659,7 +659,7 @@ const BottomSheetComponent = forwardRef<BottomSheet, BottomSheetProps>(
           status === KEYBOARD_STATUS.SHOWN &&
           position !== closedDetentPosition
         ) {
-          index = highestDetentPosition ?? DEFAULT_KEYBOARD_INDEX;
+          index = detents?.indexOf(highestDetentPosition ?? 0) ?? DEFAULT_KEYBOARD_INDEX;
```

Comment:
Just found the same bug and was about to open the same pr😁

I think it is better to just get the last item of the array since we know they are sorted in ascending order, right?

```suggestion
          index = detents?.length ? detents.length - 1 : DEFAULT_KEYBOARD_INDEX;
```